pkg/roachprod/vm/ibm/ibm_extended_types.go

```diff
@@ -485,25 +485,11 @@ func (i *instance) getPrimaryNetworkInterface() (*vpcv1NetworkInterface, error)
 	return i.networkInterface, nil
 }
 
-// isRoachprodAndValidStatus checks if a roahcprod instance in valid status.
-func (i *instance) isRoachprodAndValidStatus() (bool, string) {
+// isValidStatus checks if an instance has a valid status.
+func (i *instance) isValidStatus() (bool, string) {
 	if i == nil || i.instance == nil || i.instance.Status == nil {
```

Comment:
In `toVM`, which proceeds the call to `isRoachprodAndValidStatus`, we already have checks for :

```
	if i == nil || i.instance == nil {
		return vm.VM{
			Errors: []error{errors.New("instance is nil")},
		}
	}
```
and

```
if core.StringNilMapper(i.instance.Status) == "failed" {
		vmErrors = append(vmErrors, errors.New("instance is in failed state"))
	}
```

Any reason not to just unify the error checking in one location?